### PR TITLE
[core][state] Handle driver tasks 

### LINKED ISSUE #31575: [core][state] ray summary breaks when data is missing
### What happened + What you expected to happen

With data loss on the worker side, there might be tasks without task info (e.g. type), however ray summary assumes this to be present. 

### Versions / Dependencies

master

### Reproduction script

failure of ray summary tasks in many_tasks. 

### Issue Severity

None

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyyx <rickyx@anyscale.com>

## Why are these changes needed?
1. This PR adds additional information to the driver task event, namely, driver task type, and it's running/finished timestamps.  This allows users (i.e. the dashboard) to inspect driver task more easily. 
2. This PR also exposes the `exclude_driver` flag to state API, allowing requests through https and `ListAPiOptions` to get driver tasks, while the default behaviour from state API will still be excluding it. 
3. This PR also filters out any tasks w/o task_info to prevent missing data issue. 

## Related issue number
Closes https://github.com/ray-project/ray/issues/31575

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/experimental/state/api.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def _make_param(cls, options: Union[ListApiOptions, GetApiOptions]) -> Dict:
                 continue
 
             option_val = getattr(options, field.name)
-            if option_val:
+            if option_val is not None:
```
Comment (rickyyx) [inline python/ray/experimental/state/api.py:177]: this is a bug....

Comment (rkooo567) [reply python/ray/experimental/state/api.py:177]: what's the bug?

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
merge conflict!

File: python/ray/experimental/state/api.py (hunk 1 above)
Comment (rickyyx) [reply python/ray/experimental/state/api.py:177]: If the `option_val` is falsy, like False, or 0, it will be ignored.

**Comment by rickyyx** [comment]
Could you help stamp for the `gcs.proto` renaming cc @scv119

File: python/ray/experimental/state/api.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/experimental/state/api.py:177]: that makes sense.. 😞

**Comment by rkooo567** [comment]
Python large test seems related?

File: src/ray/protobuf/gcs_service.proto [hunk 2]
```diff
@@ -611,7 +611,7 @@ message GetTaskEventsRequest {
   // guarantee.
   optional int64 limit = 3;
   // True if task events from driver (only profiling events) should be excluded.
-  bool exclude_driver_task = 4;
+  bool exclude_driver = 4;
```
Comment (scv119) [inline src/ray/protobuf/gcs_service.proto:614]: no backward compatibility concern?

**Review by scv119 — APPROVED** [review]

File: src/ray/protobuf/gcs_service.proto (hunk 2 above)
Comment (rickyyx) [reply src/ray/protobuf/gcs_service.proto:614]: yeah, this is a really new flag we added. The code uses it is probably everything here.

File: dashboard/state_aggregator.py [hunk 3]
```diff
@@ -362,13 +362,15 @@ async def list_tasks(self, *, option: ListApiOptions) -> ListApiResponse:
         """
         job_id = None
         for filter in option.filters:
-            if filter[0] == "job_id":
+            if filter[0] == "job_id" and filter[1] == "=":
```
Comment (rickyyx) [inline dashboard/state_aggregator.py:365]: This was a bug.

Comment (rkooo567) [reply dashboard/state_aggregator.py:365]: Mind adding a unit test that can fail this?

**Comment by rkooo567** [comment]
there's merge conflict

File: dashboard/state_aggregator.py (hunk 3 above)
Comment (rickyyx) [reply dashboard/state_aggregator.py:365]: The new unit test I added actually exposes it.

**Comment by rkooo567** [comment]
lmk when we can merge it!

**Comment by rickyyx** [comment]
Just waiting for the test results to be in.

**Comment by rkooo567** [comment]
There's a merge conflict!

**Comment by rkooo567** [comment]
Failed test seems to be flaky in the master

**Pull request merged**